src/pg_query_summary_statement_type.c

```diff
@@ -0,0 +1,798 @@
+		case T_Group:
+		case T_Agg:
+		case T_WindowAgg:
+		case T_Unique:
+		case T_Gather:
+		case T_GatherMerge:
+		case T_Hash:
+		case T_SetOp:
+		case T_LockRows:
+		case T_Limit:
+		case T_PlanRowMark:
+		case T_PartitionPruneInfo:
+		case T_PartitionedRelPruneInfo:
+		case T_PartitionPruneStepOp:
+		case T_PartitionPruneStepCombine:
+		case T_PlanInvalItem:
+		case T_ExprState:
+		case T_IndexInfo:
+		case T_ExprContext:
+		case T_ReturnSetInfo:
+		case T_ProjectionInfo:
+		case T_JunkFilter:
+		case T_OnConflictSetState:
+		case T_MergeActionState:
+		case T_ResultRelInfo:
+		case T_EState:
+		case T_WindowFuncExprState:
+		case T_SetExprState:
+		case T_SubPlanState:
+		case T_DomainConstraintState:
+		case T_ResultState:
+		case T_ProjectSetState:
+		case T_ModifyTableState:
+		case T_AppendState:
+		case T_MergeAppendState:
+		case T_RecursiveUnionState:
+		case T_BitmapAndState:
+		case T_BitmapOrState:
+		case T_ScanState:
+		case T_SeqScanState:
+		case T_SampleScanState:
+		case T_IndexScanState:
+		case T_IndexOnlyScanState:
+		case T_BitmapIndexScanState:
+		case T_BitmapHeapScanState:
+		case T_TidScanState:
+		case T_TidRangeScanState:
+		case T_SubqueryScanState:
+		case T_FunctionScanState:
+		case T_ValuesScanState:
+		case T_TableFuncScanState:
+		case T_CteScanState:
+		case T_NamedTuplestoreScanState:
+		case T_WorkTableScanState:
+		case T_ForeignScanState:
+		case T_CustomScanState:
+		case T_JoinState:
+		case T_NestLoopState:
+		case T_MergeJoinState:
+		case T_HashJoinState:
+		case T_MaterialState:
+		case T_MemoizeState:
+		case T_SortState:
+		case T_IncrementalSortState:
+		case T_GroupState:
+		case T_AggState:
+		case T_WindowAggState:
+		case T_UniqueState:
+		case T_GatherState:
+		case T_GatherMergeState:
+		case T_HashState:
+		case T_SetOpState:
+		case T_LockRowsState:
+		case T_LimitState:
+		case T_IndexAmRoutine:
+		case T_TableAmRoutine:
+		case T_TsmRoutine:
+		case T_EventTriggerData:
+		case T_TriggerData:
+		case T_TupleTableSlot:
+		case T_FdwRoutine:
+		case T_Bitmapset:
+		case T_ExtensibleNode:
+		case T_ErrorSaveContext:
+		case T_IdentifySystemCmd:
+		case T_BaseBackupCmd:
+		case T_CreateReplicationSlotCmd:
+		case T_DropReplicationSlotCmd:
+		case T_AlterReplicationSlotCmd:
+		case T_StartReplicationCmd:
+		case T_ReadReplicationSlotCmd:
+		case T_TimeLineHistoryCmd:
+		case T_UploadManifestCmd:
+		case T_SupportRequestSimplify:
+		case T_SupportRequestSelectivity:
+		case T_SupportRequestCost:
+		case T_SupportRequestRows:
+		case T_SupportRequestIndexCondition:
+		case T_SupportRequestWFuncMonotonic:
+		case T_SupportRequestOptimizeWindowClause:
+		case T_Integer:
+		case T_Float:
+		case T_Boolean:
+		case T_String:
+		case T_BitString:
+		case T_ForeignKeyCacheInfo:
+		case T_IntList:
+		case T_OidList:
+		case T_XidList:
+		case T_AllocSetContext:
+		case T_GenerationContext:
+		case T_SlabContext:
+		case T_BumpContext:
+		case T_TIDBitmap:
+		case T_WindowObjectData:
+			break;
+	}
+
```

Comment:
If we invented a function that checks whether a given node is handled by the treewalker, we can call that here, and `return false` if its not supported (that way the code above is simplified, and we can share code between `pg_query_summary.c` and `pg_query_summary_statement.c`)